### PR TITLE
Validate dates before 1920

### PR DESCRIPTION
**Pull request by vassyz**
### Context
Added to date of birth, job start date, job end date pages.

Screenshot 2023-03-03 at 14 14 17

### Link to Trello card

https://trello.com/c/uZAhKWwY/1231-employer-form-validate-dates-before-1920

### CODE REVIEW

File: app/validators/date_validator.rb [hunk 1]
```diff
@@ -78,22 +78,19 @@ def validate_each(record, attribute, _value)
       return false
     end
 
-    return true unless options[:date_of_birth] || !options[:in_the_future]
-
-    if year > Time.zone.today.year
+    if (options[:past_century] || !options[:in_the_future]) &&
```
Comment (paulodeon) [inline app/validators/date_validator.rb:81]: Seems a bit confusing to mix up the two validation options here, can that be avoided?

Also it seems that `:in_the_future` actually means in the past which is pretty confusing!

File: app/validators/date_validator.rb [hunk 2]
```diff
@@ -78,22 +78,19 @@ def validate_each(record, attribute, _value)
...
       record.errors.add(attribute, :in_the_future)
       return false
     end
 
-    return true unless options[:date_of_birth]
-
-    if year < 1900
-      record.errors.add(attribute, :born_after_1900)
+    if options[:past_century] && year < 1920
+      record.errors.add(attribute, :before_century)
```
Comment (paulodeon) [inline app/validators/date_validator.rb:88]: I would prefer if the keys for validation and translation matched here

File: config/locales/en.yml [hunk 3]
```diff
@@ -84,8 +84,8 @@ en:
           attributes:
             date_of_birth:
               blank: Enter their date of birth
-              born_after_1900: Enter a year later than 1900
-              inclusion: You must be 16 or over to use this service
+              before_century: Their date of birth must be the same as or after 1 January 1920
```
Comment (paulodeon) [inline config/locales/en.yml:87]: This seems a little wordy, worth getting Daniel or Adam to check?

File: spec/validators/date_validator_spec.rb [hunk 4]
```diff
@@ -97,7 +121,7 @@ def stub_validatable_class(name, options: true)
       it { is_expected.to be_invalid }
     end
 
-    context "with max age date of birth params" do
+    context "with year fefore 1920" do
```
Comment (paulodeon) [inline spec/validators/date_validator_spec.rb:124]: Spelling

File: config/locales/en.yml (hunk 3 above)
Comment (vassyz) [reply config/locales/en.yml:87]: Text came from @adamsilver

**Review by paulodeon — APPROVED** [review]

**Pull request merged**